sdk/storage/azure-storage-common/test/ut/connection_reuse_test.cpp

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include "test_base.hpp"
+
+#include <azure/storage/blobs/block_blob_client.hpp>
+
+#include <unordered_map>
+
+namespace Azure { namespace Storage { namespace Test {
+
+  class ConnectionReuseTest : public StorageTest {
+  };
+
+  void updateHitCountPerServerMap(
+      const Azure::Core::CaseInsensitiveMap& headers,
+      std::unordered_map<std::string, int32_t>& hitCountPerServerMap)
+  {
+    // The third part of a storage request id means the server node id.
+    // For Example, RequestId:3bcf963b-601e-0054-1f40-910c39000000, '0054' is the server node which
+    // served this request.
+    std::string serverId = headers.find("x-ms-request-id")->second.substr(14, 4);
+    if (hitCountPerServerMap.count(serverId) == 0)
+    {
+      hitCountPerServerMap[serverId] = 1;
+    }
+    else
+    {
+      ++hitCountPerServerMap[serverId];
+    }
+  }
+
+  // If connection is reused, the requests with the same connection should hit the same sever. So
+  // this test verifies whether a series of requests hit the same server.
+  TEST_F(ConnectionReuseTest, IsConnectionReused_LIVEONLY_)
+  {
+    const std::string containerName = LowercaseRandomString();
+    const std::string blobName = LowercaseRandomString();
+
+    auto clientOptions = InitStorageClientOptions<Azure::Storage::Blobs::BlobClientOptions>();
+    auto containerClient1 = Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
+        StandardStorageConnectionString(), containerName + "1", clientOptions);
+    auto containerClient2 = Azure::Storage::Blobs::BlobContainerClient::CreateFromConnectionString(
+        StandardStorageConnectionString(), containerName + "2", clientOptions);
```

Comment:
why do we need to create two container clients? Do you intent to test the case that requests from two clients that don't share the same pipeline instance also reuse the same connection?

Reply:
Yeah, I think we should cover this scenario in this test case.